### PR TITLE
Introduced API for updating Build custom fields values

### PR DESCRIPTION
**Pull request by mazimkhan**
New API `updateBuildCustomFields` for updating build custom field values. 

API signature:

```
public Map<String, Object> updateBuildCustomFields(Integer buildId, Integer testProjectId, Integer testPlanId, Map<String, String> customFields)
```

Corresponding TestLink XML-RPC Code will be shared shortly.

Related TestLink PR https://github.com/kinow/testlink-java-api/pull/71

### CODE REVIEW

**Comment by kinow** [comment]
Do you have the ticket link for the TestLink XML-RPC code, please @mazimkhan ? I'll keep an eye on that and prepare to merge/release after the PHP code has been updated.

Thanks a lot for the pull request.

**Comment by mazimkhan** [comment]
Hi, yes it is here https://github.com/TestLinkOpenSourceTRMS/testlink-code/pull/47

**Comment by mazimkhan** [comment]
@PrzemekWirkus @ConorPKeegan

**Comment by mazimkhan** [comment]
@kinow we have some execute time custom fields that we want to update via junit result seeker. I see ```TestLinkAPI.reportTCResult``` API has support for sending custom field values: param ```Map<String, String> customFields```. But hudson.tasks.junit.CaseResult does not have any means to get extra attributes or additional elements. 

Can you please help if you know a solution to update execute time custom fields.

**Comment by kinow** [comment]
IIRC, in the tap-plugin I had to create a wrapper around junit-plugin objects (which were part of the core at that time), in order to add metadata collected from TAP resuls + YAMLish. Maybe that's one way to solve it.

**Comment by mazimkhan** [comment]
Thanks for your input @kinow. I will try a similar solution in testlink-plugin.

**Comment by mazimkhan** [comment]
https://github.com/TestLinkOpenSourceTRMS/testlink-code/pull/47 has been merged so this should be good to go.

**Comment by kinow** [comment]
All good to merge @mazimkhan ! Will merge it once the code has been released (will check if it's in 1.9.15).

**Comment by mazimkhan** [comment]
Awesome thanks!

**Comment by kinow** [comment]
Not a problem. Just checked that it is not in TestLink 1.9.15. Let's wait for the next version to merge and release it :-)

File: src/main/java/br/eti/kinoshita/testlinkjavaapi/constants/TestLinkMethods.java [hunk 1]
```diff
@@ -29,60 +29,65 @@
...
-                                                                                                                                                                                                                                                                                                                                                                                                                                    "tl.removePlatformFromTestPlan"), ADD_PLATFORM_TO_TEST_PLAN(
-                                                                                                                                                                                                                                                                                                                                                                                                                                            "tl.addPlatformToTestPlan");
```
Comment (kinow) [inline src/main/java/br/eti/kinoshita/testlinkjavaapi/constants/TestLinkMethods.java]: Yuck. I hadn't notice it got all that messed up. Thakns for fixing the formatting here.

**Comment by kinow** [comment]
Thanks! Preparing a 1.9.16-0 today, unless I find some issue, then will cut a release during the next week.

**Pull request merged**